### PR TITLE
Add verification scripts

### PR DESCRIPTION
**Pull request by soltysh**
@deads2k @sttts ptal

### CODE REVIEW

**Comment by soltysh** [comment]
[(3) Add verify script for client-go and api](https://trello.com/c/AFDpNB5h/1098-3-add-verify-script-for-client-go-and-api)

File: hack/update-swagger-docs.sh [hunk 1]
```diff
@@ -45,27 +43,7 @@ EOF
...
-kube::swagger::gen_types_swagger_doc "route/v1" "route/v1"
-rm -f "${SCRIPT_ROOT}/security/v1/types_swagger_doc_generated.go"
-kube::swagger::gen_types_swagger_doc "security/v1" "security/v1"
-rm -f "${SCRIPT_ROOT}/template/v1/types_swagger_doc_generated.go"
-kube::swagger::gen_types_swagger_doc "template/v1" "template/v1"
-rm -f "${SCRIPT_ROOT}/user/v1/types_swagger_doc_generated.go"
-kube::swagger::gen_types_swagger_doc "user/v1" "user/v1"
+for gv in ${API_GROUP_VERSIONS}; do
+  rm -f "${SCRIPT_ROOT}/${gv}/types_swagger_doc_generated.go"
+  kube::swagger::gen_types_swagger_doc "${gv}" "${gv}"
```
Comment (sttts) [inline hack/update-swagger-docs.sh]: :+1:

File: .travis.yml [hunk 2]
```diff
@@ -3,10 +3,17 @@ language: go
 go:
   - 1.8
 
+install:
+  - wget https://github.com/google/protobuf/releases/download/v3.0.2/protoc-3.0.2-linux-x86_64.zip
```
Comment (sttts) [inline .travis.yml]: do we check that we have the right protoc in the path?

Comment (sttts) [reply .travis.yml]: oh, this is travis, not Makefile. nvmd

File: hack/lib/init.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# This script is meant to be the entrypoint for OpenShift Bash scripts to import all of the support
+# libraries at once in order to make Bash script preambles as minimal as possible. This script recur-
+# sively `source`s *.sh files in this directory tree. As such, no files should be `source`ed outside
+# of this script to ensure that we do not attempt to overwrite read-only variables.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+API_GROUP_VERSIONS="apps/v1 authorization/v1 build/v1 image/v1 network/v1 oauth/v1 project/v1 quota/v1 route/v1 security/v1 template/v1 user/v1"
```
Comment (sttts) [inline hack/lib/init.sh]: nit: list with newlines would be easier to read.

File: hack/lib/init.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (sttts) [inline hack/lib/init.sh]: nit: technically not necessary.

File: hack/update-deepcopy.sh [hunk 5]
```diff
@@ -1,14 +1,15 @@
...
 
-set -o errexit
-set -o nounset
-set -o pipefail
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
 SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
 CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../../../k8s.io/code-generator)}
 
+verify="${VERIFY:-}"
```
Comment (sttts) [inline hack/update-deepcopy.sh]: in kube we have hack/update-codegen.sh and hack/verify-codegen.sh. Would be more self-explaining maybe.

**Comment by deads2k** [comment]
@soltysh I synced with the rebase branch and this now conflicts.  Sorry.

**Comment by soltysh** [comment]
> @soltysh I synced with the rebase branch and this now conflicts. Sorry.

No worries, I'll update it later today.

File: hack/update-deepcopy.sh (hunk 5 above)
Comment (soltysh) [reply hack/update-deepcopy.sh]: Yeah, but codegen in kube generates more than just deepcopies. When we'll do the same we can change. For now this make it explicit it's just the deepcopy.

File: hack/lib/init.sh (hunk 3 above)
Comment (soltysh) [reply hack/lib/init.sh]: 👍 good point, updated both envs.

File: hack/lib/init.sh (hunk 4 above)
Comment (soltysh) [reply hack/lib/init.sh]: Copied over from origin, let it stay ;)

**Comment by soltysh** [comment]
Rebased and some nits addressed. Ready for merge on green travis.

File: hack/update-swagger-docs.sh (hunk 1 above)
Comment (deads2k) [reply hack/update-swagger-docs.sh]: yay!

**Comment by deads2k** [comment]
> Rebased and some nits addressed. Ready for merge on green travis.

Thanks

**Pull request merged**